skopt/dummy_opt.py

```diff
@@ -52,14 +70,32 @@ def dummy_minimize(func, dimensions, n_calls=100, random_state=None):
     """
     rng = check_random_state(random_state)
     space = Space(dimensions)
-    X = space.rvs(n_samples=n_calls, random_state=rng)
-
-    init_y = func(X[0])
+    if x0 is None:
+        x0 = []
+    x0 = list(x0)
+    if x0 and not isinstance(x0[0], list):
+        x0 = [x0]
+    if y0 is None:
+        y0 = [func(x) for x in x0]
+        n_calls -= len(y0)
+    if isinstance(y0, Iterable):
+        y0 = list(y0)
+    else:
+        y0 = [y0]
```

Comment:
Can we restructure like this?

```python
if y0 is None and x0:
    y0 = [func(x) for x in x0]
    n_func_calls = len(y0)
elif x0:
    if instance(y0, Iterable):
        y0 = list(y0)
    elif isinstance(y0, numbers.Number):
        y0 = [y0]
     else:
        raise ValueError
     if len(x0) != len(y0):
         raise ValueError
```

And do explicitly do `n_samples = n_calls - len(y0)` below, so that it a bit easier to read.